src/Compiler/Service/ServiceParsedInputOps.fs

```diff
@@ -47,6 +47,15 @@ type RecordContext =
     | New of path: CompletionPath * isFirstField: bool
     | Declaration of isInIdentifier: bool
 
+[<RequireQualifiedAccess>]
+type MatchContext =
+    /// Completing an identifier in a match clause pattern
+    /// (e.g. 'match x with Som| y -> ...')
+    | ClausePatternIdentifier of range: range
+
+    /// Completing a match clause guard (e.g. match x with Some y when y| -> ...)
+    | ClauseGuard
```

Comment:
Have you checked that completions at active pattern arguments are still ok?

Reply:
Yep, `ClausePatternIdentifier` is used when the caret is within the range of the outermost identifier of the match clause. Case/active pattern argument completions remain unchanged.

```fsharp
match x: int option with
| Som$ x // will only show `Option` cases and compatible active patterns
| Some x$ // maintains status quo and shows an unfiltered completion list
```

I have also tried extending the behavior to arguments:

```fsharp
match x: int option option with
| Some (N$) // will only show `Option` cases and compatible active patterns
```

It appears to work fine, but I'd rather get this PR in as is and then proceed to make further improvements.